johnzon-core/src/main/java/org/apache/johnzon/core/Snippet.java

```diff
@@ -0,0 +1,413 @@
+                return;
+            }
+
+            generator.writeStartArray();
+            for (final JsonValue jsonValue : array) {
+                if (terminate()) {
+                    break;
+                }
+                write(jsonValue);
+            }
+            generator.writeEnd();
+        }
+
+        private void write(final JsonObject object) {
+            if (object.isEmpty()) {
+                generator.write(object);
+                return;
+            }
+
+            generator.writeStartObject();
+            for (final Map.Entry<String, JsonValue> entry : object.entrySet()) {
+                if (terminate()) {
+                    break;
+                }
+                write(entry.getKey(), entry.getValue());
+            }
+            generator.writeEnd();
+        }
+
+        private void write(final String name, final JsonValue value) {
+            switch (value.getValueType()) {
+                case ARRAY:
+                    generator.writeStartArray(name);
+                    final JsonArray array = value.asJsonArray();
+                    for (final JsonValue jsonValue : array) {
+                        if (terminate()) {
+                            break;
+                        }
+                        write(jsonValue);
+                    }
+                    generator.writeEnd();
+
+                    break;
+                case OBJECT:
+                    generator.writeStartObject(name);
+                    final JsonObject object = value.asJsonObject();
+                    for (final Map.Entry<String, JsonValue> keyval : object.entrySet()) {
+                        if (terminate()) {
+                            break;
+                        }
+                        write(keyval.getKey(), keyval.getValue());
+                    }
+                    generator.writeEnd();
+
+                    break;
+                default: {
+                    generator.write(name, value);
+                }
+            }
+        }
+
+        private boolean terminate() {
+            flush.run();
+            return snippet.terminate();
+        }
+
+        private String get() {
+            generator.flush();
+            return snippet.isTruncated() ? snippet.get() + "..." : snippet.get();
+        }
+
+        @Override
+        public void close() {
+            generator.close();
+        }
+
+        /**
+         * Specialized Writer with three internal states:
+         * Writing, Completed, Truncated.
+         *
+         * When there is still space left for more json, the
+         * state will be Writing
+         *
+         * If the last write brought is exactly to the end of
+         * the max length, the state will be Completed.
+         *
+         * If the last write brought us over the max length, the
+         * state will be Truncated.
+         */
+        class SnippetWriter extends Writer implements Buffered {
```

Comment:
think the move from stream to writer looses the encoding so snippet will be broken, no?

Reply:
Resolving as this was discussed on the list.